firmware/robot2015/src-ctrl/modules/motors/motors.hpp

```diff
@@ -0,0 +1,74 @@
+/**
+ * RoboJackets: RoboCup SSL Firmware
+ *
+ * Copyright (C) 2015 RoboJackets JJ
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, in version 3.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * motors.hpp - BLDC data structures.
+ */
+
+#pragma once
+
+
+// Includes
+#include <cstdint>
+#include <array>
+#include <vector>
+
+
+/* Any math using the delta values for the hall/encoder (we really
+ * only care about encoder readings...) is best handled through a task
+ * queue where the calculations can be managed on the kernel's scheduler.
+ * So we'll leave the MOTOR_MAX_SAMPLES at a minimum in the struct.
+ */
+#define MOTOR_MAX_SAMPLES 2
+
+
+typedef uint16_t motorVel_t;
+
+struct motorErr_t {
+	/*
+	Available Flags:
+	=================
+	GVDD_OV
+	GVDD_UV
+	PVDD_UV
+	OTSD
+	OTW
+	FETHA_OC
+	FETLA_OC
+	FETHB_OC
+	FETLB_OC
+	FETHC_OC
+	FETLC_OC
+	*/
+	bool encOK;
+	bool hallOK;
+	std::array<uint16_t, 2> drvStatus;
+};
+
+
+struct motor_t {
+	motorVel_t 	targetVel;
```

Comment:
The LQR motion control stuff I have setup calculates a voltage to send to each motor rather than a velocity.  I'll run you through the controls once we're all back at Tech (and we can change it around if needed), but the basic idea is that if the controller takes into account the whole robot's acceleration, it can do a better job if it calculates voltages than velocities.